### PR TITLE
SDM generator

### PR DESCRIPTION
**Pull request by jessitron**
this adds a new generator based on github-sdm.

The bit about finding the author doesn't work, graphclient returns a 400, but that's a different problem.

### CODE REVIEW

**Review by ddgenome — CHANGES_REQUESTED** [review]
The logzio stuff is very much used.

**Comment by ddgenome** [comment]
This should fix your logzio compilation problems: https://github.com/atomist/k8-automation/blob/3a0c01eb218cccbe18e42e7de5a90f934949666a/src/index.d.ts#L43

**Review by ddgenome — APPROVED** [review]

File: config/default.json [hunk 1]
```diff
@@ -1,10 +1,10 @@
 {
   "endpoints": {
-    "api":"https://automation-staging.atomist.services/registration",
-    "graphql": "https://automation-staging.atomist.services/graphql/team"
+    "api": "https://automation.atomist.com/registration",
+    "graphql": "https://automation.atomist.com/graphql/team"
```
Comment (ddgenome) [inline config/default.json]: These changes will break the staging.conf. Did you mean to commit these? I typically create a new conf and use that to test in a specific team.

File: config/default.json [hunk 2]
```diff
@@ -1,10 +1,10 @@
...
   },
   "groups": null,
-  "teamIds": "T1L0VDKJP",
+  "teamIds": "T1JVCMVH7",
```
Comment (ddgenome) [inline config/default.json]: Intentional?

File: src/commands/editor/node/updatePackageLock.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import { logger } from "@atomist/automation-client";
+import { LocalProject } from "@atomist/automation-client/project/local/LocalProject";
+import { Project } from "@atomist/automation-client/project/Project";
+import * as child_process from "child_process";
+
+export async function updatePackageLock(project: Project) {
+    const result = new Promise((resolve, reject) => child_process.exec("npm install",
```
Comment (ddgenome) [inline src/commands/editor/node/updatePackageLock.ts]: This seems like a lot of compute/network just to update the package-lock.json. Couldn't we just update the package name and version manually?

**Review by ddgenome — COMMENTED** [review]
A couple questions.

File: config/default.json (hunk 2 above)
Comment (jessitron) [reply config/default.json]: nope, that's satellite-of-love. better to leave it as community

File: src/commands/editor/node/updatePackageLock.ts (hunk 3 above)
Comment (jessitron) [reply src/commands/editor/node/updatePackageLock.ts]: did it have a package-lock before? 
... yeah, I guess it does
is it accurate?
... maybe

so, probably. But the way I update package-lock.json is by running npm install, so I wanted the automation to do that.

It also has the effect of applying formatting to package.json, which masks a bug in client that messed up the spacing when we changed that.

Comment (jessitron) [reply src/commands/editor/node/updatePackageLock.ts]: (I fixed the bug in client but it isn't released afaik)

**Pull request merged**